### PR TITLE
Reorganize `scripts` folder to prepare for packaging JS scripts

### PR DESCRIPTION
**Pull request by Eric-Arellano**
Some projects are going to start using our JS scripts soon. To help us with packaging the scripts, this PR bundles all of the JS into `scripts/js` whereas before it was spread out over `scripts/commands` and `scripts/lib`.

Beyond packaging the scripts, I think this is a cleaner organization anyways to differentiate between our three scripts projects:

* `js`
* `nb-tester`
* `pr-previews` (renamed from `pr_previews`)

(In a follow up, we might want to centralize all of our config into `scripts/config`)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,23 +5,23 @@
   "author": "Qiskit Development Team",
   "license": "Apache-2.0",
   "scripts": {
-    "check": "npm run check:qiskit-bot && npm run check:patterns-index && npm run check:metadata && npm run check:spelling && npm run check:internal-links && npm run check:orphan-pages && npm run check:fmt",
```
Comment (Eric-Arellano) [inline package.json:8]: I moved these user-facing scripts below our dev-facing scripts `fmt`, `test`, and `typecheck`

File: scripts/js/commands/checkStaleImages.ts [hunk 2]
```diff
@@ -10,8 +10,6 @@
 // copyright notice, and modified files need to carry a notice indicating
 // that they have been altered from the originals.
 
-// To run:
-//   node -r esbuild-register scripts/commands/find-stale-images.ts
```
Comment (Eric-Arellano) [inline scripts/js/commands/checkStaleImages.ts:14]: You can use `npm run check:stale-images`

File: scripts/js/lib/fs.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ import path from "path";
 import { $ } from "zx/core";
 
 export function getRoot() {
-  return path.normalize(`${__dirname}/../../`);
+  return path.normalize(`${__dirname}/../../../`);
 }
```
Comment (Eric-Arellano) [inline scripts/js/lib/fs.ts:20]: This function is going to need to be reworked in preparation for packaging the repository. I plan to tackle this next week.

**Review by frankharkins — APPROVED** [review]
Thanks!

**Pull request merged**